build-system/compile/internal-version.js

```diff
@@ -26,40 +26,77 @@ const argv = minimist(process.argv.slice(2), {
+  //   `master`'s commit history, then the current commit is <C>).
+  // - Find the commit time of <C> (<C>.time). Note that commit time might be
+  //   different from author time! e.g., commit time might be the time that a PR
+  //   was merged into `master`, or a commit was cherry-picked onto the brabnch;
+  //   author time is when the original author of the commit ran the
+  //   "git commit" command.
+  // - The version number is <C>.time.format("YYmmDDHHMM", "UTC") + <X> (the
+  //   pseudo-code assumes standard `.strftime` formatting, and <X> gets
+  //   zero-padded until it is three digits long).
+  //   - The maximum number of cherry-picks in a single branch is 999. More than
+  //     that will fail to build. This should never happen.
+  //
+  // Examples:
+  // 1. The version number of a release built from the HEAD commit on `master`,
+  //    where that HEAD commit was committed on April 25, 2020 2:31 PM EDT would
+  //    be `2004251831000`.
+  //    - EDT is UTC-4, so the hour value changes from EDT's 14 to UTC's 18.
+  //    - The last three digits are 000 as this commit is on `master`.
+  //
+  // 2. The version number of a release built from a local working branch (e.g.,
+  //    on a developer's workstation) that was split off from a `master` commit
+  //    from May 6, 2021 10:40 AM PDT and has multiple commits that exist only
+  //    on local working branch would be `2105061840000`.
+  //    - PDT is UTC-7, so the hour value changes from PDT's 10 to UTC's 17.
+  //    - The last three digits are 000 as this commit is on a branch that was
+  //      split from `master`, and does not have any cherry-picked commits.
+  //
+  // 3. For a release built from a local working branch that was split off from
+  //    a `master` commit from November 9, 2021 11:48 PM PST, and then:
+  //    - had one commit that was cherry-picked from `master`,
+  //    - followed by two commits that were created directly on the branch, the
+  //      last of which was commited on November 23, 2021 6:38 PM PST,
+  //    - followed by twelve commits that were cherry-picked from `master`,
+  //    then its version number would be `2111240238014`.
+  //    - The latest twelve commits are cherry-picks from `master`, and the one
+  //      before them is not, so our last three digits are set to 014.
+  //    - PST is UTC-8, so the hour value changes from PST's 18 to UTC's 2, and
+  //      one day is added.
+  let numberOfCherryPicks = 0;
+  const commitCherriesInfo = gitCherryMaster().reverse();
+  for (const {isCherryPick} of commitCherriesInfo) {
+    if (!isCherryPick) {
+      break;
     }
-    return `${lastCommitFormattedTime}${lastDigit}`;
+    numberOfCherryPicks++;
+  }
+  if (numberOfCherryPicks > 999) {
+    throw new Error(
+      `This branch has ${numberOfCherryPicks} cherry-picks, which is more than 999, the maximum allowed number of cherry-picks!`
+    );
   }
+
+  const lastCommitFormattedTime = gitCommitFormattedTime(
```

Comment:
This is the only place `gitCommitFormattedTime()` is used. Could you revise the date format to omit `%S` and avoid the `.slice(0, -2)`?

Reply:
I thought about it, but I prefer to keep the original in case someone else will use the other function expecting the existing behaviour